Update client libraries

NicMcPhee opened the pull request:
This updated a bunch of the client library dependencies and switches over to the new `@cypress/schematics` Cypress library.

NicMcPhee committed:
Angular CLI update for packages - @angular-eslint/schematics@latest, @angular/cdk@latest, @angular/cli@latest, @angular/core@latest, @angular/material@latest

NicMcPhee committed:
Remove unnecessary `docker-build` flags in `package.json`

NicMcPhee committed:
Update rxjs

NicMcPhee committed:
Uninstall old `istanbul` coverage packages

NicMcPhee committed:
Fix location or `karma-coverage` in `package.json`

NicMcPhee committed:
Update a bunch of client packages & convert to `@cypress/schematic`

NicMcPhee committed:
Fix a race condition in cypress/.../app.spec.ts 

This is lifted from a commit on the iteration template by @helloworld12321:

> We were clicking the "open sidenav" button, and then expecting the
"open sidenav" button to be visible. This will *only* be the case for a
split-second before the sidenav appears.

> I would guess that this has always been an issue, but it's started
failing consistently on GitHub actions as of Cypress version 9.

https://github.com/UMM-CSci-3601/3601-iteration-template/commit/d33308eab5d6213885d1aec626a73e4f72fd7551

Co-authored-by: Joe Walbran <walbr037@morris.umn.edu>

floogulinc commented on client/package.json:
```diff
@@ -9,57 +9,55 @@
     "e2e": "ng e2e",
     "cy:open": "cypress open",
-    "docker-build": "npm run build -- --configuration production"
+    "docker-build": "npm run build"
```
If you are going to change this you need to apply the changes to `angular.json` to make production the default config. See https://github.com/UMM-CSci-3601/3601-iteration-template/commit/9234ad74aac979a048fcb1c1376cc42e4cb61c98 and https://github.com/UMM-CSci-3601/3601-lab3_angular-javalin/pull/767

floogulinc replied:
Actually why do we have this at all? the Dockerfiles were removed, this line can be removed.

NicMcPhee replied:
Yeah, I was a little confused about why the Docker stuff was here. Presumably it got carried over from the iteration template at some point.

NicMcPhee replied:
Thanks for creating the issue #35 that lists everything that needs to be removed for this!

floogulinc commented on client/angular.json:
```diff
@@ -113,7 +113,7 @@
-          "builder": "@nrwl/cypress:cypress",
+          "builder": "@cypress/schematic:cypress",
           "options": {
             "cypressConfig": "cypress.json",
             "tsConfig": "cypress/tsconfig.json",
```
See my comments in https://github.com/UMM-CSci-3601/3601-iteration-template/pull/712
```suggestion
```

floogulinc commented on client/package.json:
```diff
@@ -9,57 +9,55 @@
+    "@cypress/schematic": "^1.6.0",
+    "@types/jasmine": "~3.10.3",
+    "@types/jasminewd2": "~2.0.10",
+    "@types/node": "^17.0.18",
```
This shouldn't be upgraded to 17

floogulinc commented on client/package.json:
```diff
@@ -9,57 +9,55 @@
+    "@angular/language-service": "~13.2.3",
+    "@cypress/schematic": "^1.6.0",
+    "@types/jasmine": "~3.10.3",
+    "@types/jasminewd2": "~2.0.10",
```
This was used by Protractor and isn't needed anymore
```suggestion
```

NicMcPhee replied:
I was wondering about that, actually, since none of the other repos had it. I'd made a note to look into it, but you've saved me the trouble of figuring it out. 😀

floogulinc reviewed (changes requested):
I highly recommend you apply the changes from this lab 3 PR https://github.com/UMM-CSci-3601/3601-lab3_angular-javalin/pull/767.

NicMcPhee committed:
Several other clean-up operations

These are mostly small changes to match the state of the Cypress files in other repos.

NicMcPhee committed:
Remove some unused config/dependencies from the client

Thanks to @floogulinc for catching these.

Co-authored-by: Paul Friederichsen <floogulinc@gmail.com>

NicMcPhee committed:
Modernize `angular.json`

This is a set of changes from @floogulinc from an iteration template commit (https://github.com/UMM-CSci-3601/3601-iteration-template/commit/9234ad74aac979a048fcb1c1376cc42e4cb61c98).

Co-authored-by: Paul Friederichsen <floogulinc@gmail.com>

NicMcPhee committed:
Merge remote-tracking branch 'origin/update-client-libraries' into update-client-libraries

NicMcPhee committed:
Bring over config changes from Lab 3

This is a few remaining changes that got made by Floogulinc on Lab 3 (https://github.com/UMM-CSci-3601/3601-lab3_angular-javalin/pull/767/files) that needed to be brought over here to this lab.

Co-authored-by: Paul Friederichsen <floogulinc@gmail.com>

NicMcPhee committed:
Remove lingering Docker files

It's not clear why there were ever Docker files in this lab (perhaps copied over from the iteration template at some point?), but this cleans all that out.

Thanks to @floogulinc for making the list of things to change here.

Closes #35

Co-authored-by Paul Friederichsen <floogulinc@gmail.com>

NicMcPhee committed:
Roll back `@types/node` dependency to v16

floogulinc commented on client/package.json:
```diff
@@ -9,57 +9,54 @@
     "lint": "ng lint",
     "e2e": "ng e2e",
     "cy:open": "cypress open",
-    "docker-build": "npm run build -- --configuration production"
+    "docker-build": "npm run build"
```
This still needs to be removed.

NicMcPhee replied:
I thought I made all the associated changes so we could remove these flags. Am I missing something?

NicMcPhee committed:
Remove reference to Docker in `package.json`

Thanks to @floogulinc for catching that!

NicMcPhee committed:
Merge branch 'main' into update-client-libraries
